ndc-test/src/test_cases/query/relationships/mod.rs

```diff
@@ -0,0 +1,221 @@
+                    )
+                )
+                ;
+
+                Some(())
+            }
+        });
+    }
+
+    Some(())
+}
+
+async fn select_top_n_using_foreign_key<C: Connector>(
+    connector: &C,
+    collection_type: &models::ObjectType,
+    collection_info: &models::CollectionInfo,
+    schema: &models::SchemaResponse,
+    foreign_key_name: &str,
+    foreign_key: &models::ForeignKeyConstraint,
+) -> Result<()> {
+    let mut fields = super::common::select_all_columns(collection_type);
+
+    let other_collection = schema
+        .collections
+        .iter()
+        .find(|c| c.name == foreign_key.foreign_collection)
+        .ok_or(Error::CollectionIsNotDefined(
+            foreign_key.foreign_collection.clone(),
+        ))?;
+
+    if other_collection.arguments.is_empty() {
+        let other_collection_type = schema
+            .object_types
+            .get(other_collection.collection_type.as_str())
+            .ok_or(Error::CollectionTypeIsNotDefined(
+                other_collection.collection_type.clone(),
+            ))?;
+
+        let other_fields = super::common::select_all_columns(other_collection_type);
+
+        fields.insert(
+            foreign_key_name.into(),
+            models::Field::Relationship {
+                query: Box::new(models::Query {
+                    aggregates: None,
+                    fields: Some(other_fields.clone()),
+                    limit: Some(10),
+                    offset: None,
+                    order_by: None,
+                    predicate: None,
+                }),
+                relationship: "__relationship".into(),
+                arguments: BTreeMap::new(),
+            },
+        );
+
+        let query_request = models::QueryRequest {
+            collection: collection_info.name.clone(),
+            query: models::Query {
+                aggregates: None,
+                fields: Some(fields.clone()),
+                limit: Some(10),
+                offset: None,
+                order_by: None,
+                predicate: None,
+            },
+            arguments: BTreeMap::new(),
+            collection_relationships: BTreeMap::from_iter([(
+                "__relationship".into(),
+                models::Relationship {
+                    column_mapping: foreign_key.column_mapping.clone(),
+                    relationship_type: models::RelationshipType::Object,
+                    target_collection: foreign_key.foreign_collection.clone(),
+                    arguments: BTreeMap::new(),
+                },
+            )]),
+            variables: None,
+        };
+
+        let response = connector.query(query_request).await?;
+
+        super::expectations::expect_single_rows(&response)?;
+    } else {
+        eprintln!("Skipping parameterized relationship {}", foreign_key_name);
+    }
+
+    Ok(())
+}
+
+async fn select_top_n_using_foreign_key_as_array_relationship<C: Connector>(
```

Comment:
Love the name here 😆.

Reply:
They're only going to get better from here.

Reply:
They're only going to get better from here.